08_AdvancedTypes/Program.cs

```diff
@@ -0,0 +1,108 @@
+﻿using _08_AdvancedTypes;
+using _08_AdvancedTypes.DTO;
+using System.Text.Json;
+
+IApiDataReader starwarsAPI = new StarWarsApiDataReader();
+
+string responseContent = await starwarsAPI.ReadAsync("https://swapi.dev/api/", "planets");
+
+IEnumerable<Planet> planets =
+    JsonSerializer.Deserialize<PlanetsDTO>(responseContent).
+    results.Select(p =>
+     new Planet()
+     {
+         Name = p.name,
+         Diameter = int.TryParse(p.diameter, out int res) ? res + "" : "",
+         SurfaceWater = int.TryParse(p.surface_water, out res) ? res + "" : "",
+         Population = int.TryParse(p.population, out res) ? res + "" : ""
+     });
+
+var tablePrinter = new UniversalTablePrinter(planets);
+tablePrinter.PrintToConsole();
+
+
+Console.WriteLine("The statistics of which property would you like to see?" +
+    $"{Environment.NewLine}population" +
+    $"{Environment.NewLine}diameter" +
+    $"{Environment.NewLine}surface water");
+
+
+string desiredStat = Console.ReadLine().ToLower();
+switch (desiredStat)
+{
+    case "population":
+        Planet pMin = planets.First();
+        Planet pMax = planets.First();
+        foreach (Planet planet in planets)
```

Comment:
It is almost good that you go over each calculating both min and max together instead of traversing for each one, if not for the tryparses.
The code for getting min and max is basically the same so you could separate it in another method and have it more generic, you could pass either the property name or an expression for it to get the correct property to compare, you did reflection for the table printer you could do it here as well :), probably not performant tho